core/drivers/crypto/crypto_api/include/drvcrypt.h

```diff
@@ -0,0 +1,69 @@
+/* SPDX-License-Identifier: BSD-2-Clause */
+/*
+ * Copyright 2018-2019 NXP
+ *
+ * Brief   Crypto Driver exported constants and interfaces.
+ */
+#ifndef __DRVCRYPT_H__
+#define __DRVCRYPT_H__
+
+/* Global includes */
+#include <tee_api_types.h>
+
+#ifdef CFG_CRYPTO_DRV_DBG
+#define CRYPTO_TRACE	DMSG
+#else
+#define CRYPTO_TRACE(...)
+#endif
+
+/*
+ * Crypto Library Algorithm enumeration
+ */
+enum drvcrypt_algo_id {
+	CRYPTO_HASH = 0,     /* HASH driver */
+	CRYPTO_HMAC,         /* HMAC driver */
+	CRYPTO_MAX_ALGO      /* Maximum number of algo supported */
+};
+
+/*
+ * Cryptographic buffer type
+ */
+struct drvcrypt_buf {
+	uint8_t *data;   /* Pointer to the data buffer */
+	size_t  length;  /* Length in bytes of the data buffer */
+};
+
+/*
+ * Cryptographic module registration
+ *
+ * Inputs:
+ * idx  Crypto index in the array
+ * ops  Reference to the cryptographic module
+ *
+ * Returns:
+ * 0   Success
+ * (-1) Error
+ */
+int drvcrypt_register(enum drvcrypt_algo_id idx, void *ops);
```

Comment:
This should rather return a `TEE_Result` value.

Reply:
ok